### ISSUE TITLE
Usage Documentation Needed for tf.image.yuv_to_rgb

### ISSUE THREAD

**Issue by alexdwu13** [tensorflow/tensorflow#37067]
## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/image/yuv_to_rgb

## Description of issue (what needs changing):

The tf.image.yuv_to_rgb method specifies a YUV input of shape (H,W,3) and an RGB output of the same shape. It also notes:

> The output is only well defined if the Y value in images are in [0,1], U and V value are in [-0.5,0.5].

However YUV is natively encoded in HxWx1.5 Bytes, with values ranging from 0-255. Considering that multiple YUV-RGB conversion standards exist, it is unclear what pre-processing steps need to be done by a user who wants to pass YUV inputs to his network (https://en.wikipedia.org/wiki/YCbCr#JPEG_conversion).

### Usage example

More documentation on the proper usage of this method would be highly helpful. Specifically:

- An example showing how to pre-process a raw YUV image of size HxWx1.5B to the expected shape of (H,W,3) with normalized values Y: [0,1], UV: [-0.5,0.5].
- An example of how one might append this method to an RGB-trained model to enable it to accept YUV inputs during inference. A likely scenario might be exporting a frozen model to an Android device that natively captures in YUV.

**Comment by jvishnuvardhan** [comment]
@alexdwu13 Thanks for creating the issue. Would you like to create a PR to update the doc? Thanks!

**Comment by alexdwu13** [comment]
@jvishnuvardhan I'd be happy to create a PR -- but I myself am not clear about the usage. I've tried munging YUV images to the input format expected by `tf.image.yuv_to_rgb` but have so far not been successful https://stackoverflow.com/questions/60067415/what-is-the-correct-usage-of-tf-image-yuv-to-rgb-output-is-distorted.
Would you be able to grab ahold of the original authors of this method?

**Comment by yaoshiang** [comment]
I left a comment on the PR 37565. I think the documentation has been incorrect on the [-0.5, 0.5] range for the UV channels. And there are bigger issues with the function itself. 

1. Prior to this issue, the documentation specified that the U and V channels need to be in the range [-0.5, 0.5]. That is incorrect. The kernels in the source show that the correct ranges are 

u: -0.43601035 to + 0.43601035
v: -0.61497538 to + 0.61497538

2. The defined ranges for U and V per the ITU spec are in fact 

[-0.436, +0.436]
[-0.615, +0.615]

3. 

The PR 37565 created a linear interpolation example for the (incorrectly documented) range of [-0.5, 0.5] for the U and V channels. 

4.

PR 37565  implies that a TF developer might end up with a YUV range of [0, 256). That is extremely unlikely because the YUV space is not "square", it is "diamond" and corners like YUV= (1,1,1) are in fact possible colors. YUV=(1,1,1) would mean a "maximum bright white color with a lot of red and blue". This results in a color with more red and blue than an LCD display can render, or the human eye can perceive. 

>>> tf.image.yuv_to_rgb(tf.constant([1.0, 0.436, 0.615]))
<tf.Tensor: shape=(3,), dtype=float32, numpy=array([1.7010281, 0.4708535, 1.8859789], dtype=float32)>

Finally, the rgb_to_yuv and yuv_to_rgb functions are problematic because they assume a linear, non-gamma corrected RGB color values. The colorspace sRGB is gamma corrected, which is what the majority of images on the web or captured by a camera will be rendered in. 

--

To fix the documentation, I'd suggest a quick update to the correct UV ranges.  I'd also suggest a strong caveat that this function is likely not what people are looking for - a better transform for gamma corrected RGB (namely, sRGB) would be YCbCr. I'd also suggest we remove the example recently added of the linear transform, since it's unlikely a developer would ever get the raw YUV values from something other than the rgb_to_yuv function. 

Here is the start of some example code for yuv_to_rgb... if these suggestions make sense, I'll finish this up and make it a PR.

def yuv_to_rgb(images):
  """Converts one or more images from YUV to RGB.

  Outputs a tensor of the same shape as the `images` tensor, containing the RGB
  value of the pixels.
  The output is only well defined if the Y values in images are in [0,1],
  U values are in [-0.43601035, 0.43601035], and V value are in 
  [-0.61497538,0.61497538].


  # Test approximate range of Y, U, and V. 
  >>> rgb_images = tf.random.uniform(shape=[60000, 32, 32, 3], minval=0, maxval=256, dtype=tf.int32)
  >>> rgb_images = tf.cast(rgb_images, tf.uint8)
  >>> rgb_float_images = tf.image.convert_image_dtype(rgb_images, tf.float32)
  >>> yuv_images = tf.image.rgb_to_yuv(rgb_float_images)

  >>> # Approximate min and max for Y channel
  >>> tf.reduce_min(yuv_images[...,0])
  <tf.Tensor: shape=(), dtype=float32, numpy=0.0>
  >>> tf.reduce_max(yuv_images[...,0])
  <tf.Tensor: shape=(), dtype=float32, numpy=1.0>

  # Approximate min and max for U channel
  >>> tf.reduce_min(yuv_images[...,1])
  <tf.Tensor: shape=(), dtype=float32, numpy=-0.43543333>
  >>> tf.reduce_max(yuv_images[...,1])
  <tf.Tensor: shape=(), dtype=float32, numpy=0.43601036>

  # Approximate min and max for V channel
  >>> tf.reduce_min(yuv_images[...,2])
  <tf.Tensor: shape=(), dtype=float32, numpy=-0.6149754>
  >>> tf.reduce_max(yuv_images[...,2])
  <tf.Tensor: shape=(), dtype=float32, numpy=0.6149754>

  This function us likely not appropriate for the typical images found on the web and generated by
  digital cameras. Those render the image in the gamma corrected sRGB color space. The YUV colorspace
  was based on analog, non-gamma corrected RGB values. The YCbCr colorspace was designed for the 
  sRGB colorspace and may be more appropriate.
